DirectXTex/DirectXTexTGA.cpp

```diff
@@ -258,6 +264,59 @@ namespace
     }
 
 
+    //-------------------------------------------------------------------------------------
+    // Reads palette for paletted TGA formats
+    //-------------------------------------------------------------------------------------
+    HRESULT ReadPalette(
+        uint8_t* header,
+        _In_reads_bytes_(size) const void* pSource,
+        size_t size,
+        uint8_t palette[256 * 4]) noexcept
+    {
+        using namespace DirectX;
+
+        auto pHeader = reinterpret_cast<const TGA_HEADER*>(header);
+
+        if (pHeader->bColorMapType == 0 || pHeader->wColorMapLength == 0)
+        {
+            return S_OK;
+        }
+
+        if (pHeader->wColorMapLength > 256)
+        {
+            return HRESULT_E_NOT_SUPPORTED;
+        }
+        if (size_t(pHeader->wColorMapLength * (pHeader->bColorMapSize / 8)) > size)
+        {
+            return HRESULT_E_INVALID_DATA;
+        }
+
+        const uint8_t* bytes = reinterpret_cast<const uint8_t*>(pSource);
```

Comment:
``auto`` is best in situations like there where you are just replicating the type on LHS and RHS.